test/file01.ts

```diff
@@ -0,0 +1,45 @@
+import { CdnDto } from '@xsite-ui/common.cdn';
+import { buildImageSrc } from '@xsite-ui/common.image';
+import { CrossOffersChart } from '@xsite-ui/common.rapido';
+import { OfferProduct } from '@xsite-ui/fintech.shared-types';
+import { getProductIdentifier, isApplicationEntityDataType } from '@xsite-ui/fintech.utils';
+
+import { BucketProps } from '../types';
+
+export function getOffersCrossChartProps(
+  crossOffersChart: BucketProps['crossOffersChart'],
+  cdnData: CdnDto,
+) {
+  const {
+    title = '',
+    secondTitle = '',
+    subTitle = '',
+    numOfOffers = 3,
+    productListV3 = {},
+  } = crossOffersChart || {};
+
+  if (!isApplicationEntityDataType<CrossOffersChart>(crossOffersChart)) {
+    return null;
+  }
```

Comment:
:robot: OpenAI

The check for `isApplicationEntityDataType` should be done before accessing any properties of `crossOffersChart`. Move this check to the beginning of the function.

<!-- This is an auto-generated comment by OpenAI -->